comb_spec_searcher/comb_spec_searcher.py

```diff
@@ -420,18 +425,47 @@ def status(self) -> str:
         status += self.classdb.status() + "\n"
         status += self.classqueue.status() + "\n"
         status += self.ruledb.status() + "\n"
-        status += self.gc_status()
+        status += self.mem_status(final_status=final_status)
         return status
 
     @cssmethodtimer("status")
-    def gc_status(self) -> str:  # pylint: disable=no-self-use
+    def mem_status(self, final_status: bool) -> str:  # pylint: disable=no-self-use
         # The self argument is actually used by the decorator
         status = "Memory Status:\n"
         status += "\tTotal Memory OS has Allocated: {}\n".format(
             size_to_readable(get_mem())
         )
         if platform.python_implementation() == "CPython":
-            pass
+            if final_status or (
+                time.time() - self.last_pympler_start > 100 * self.last_pympler_duration
+            ):
```

Comment:
should the timing be isolated to the auto_search function - perhaps you are running this in your shell for debugging purposes

Reply:
I can do that. It will mean checking again whether we're in cpython or pypy (because the mem timing only matters for cpython)

Reply:
You could make it an "elaborate" status update in general. There are other things we might want to check that take some time to compute, e.g. number of rules upto equivalence, number of equivalent classes etc etc, that could be added to the elaborate thing. Then you'd only check if in cpython or pypy before calling the functions that depend on that.

Reply:
I think I get what you're saying. Will attempt this.